Plugin/CppApi/source/CoordinateConversionController.cpp

```diff
@@ -27,11 +42,66 @@ namespace Toolkit
 
 using CoordinateType = CoordinateConversionOptions::CoordinateType;
 
+/*!
+  \brief Default constructor with an optional \a parent.
+ */
 CoordinateConversionController::CoordinateConversionController(QObject* parent):
   AbstractTool(parent)
 {
   ToolManager::instance().addTool(this);
 
+  const auto setupGeoView = [this](GeoView* geoView) -> bool
+  {
+    if (!geoView)
+      return false;
+
+    if (dynamic_cast<Esri::ArcGISRuntime::SceneQuickView*>(geoView))
+    {
+      auto sceneView = static_cast<Esri::ArcGISRuntime::SceneQuickView*>(geoView);
+
+      setSpatialReference(sceneView->spatialReference());
+      connect(sceneView, &Esri::ArcGISRuntime::SceneQuickView::spatialReferenceChanged, this,
+      [sceneView, this]()
+      {
+        setSpatialReference(sceneView->spatialReference());
+      });
+
+      connect(sceneView, &Esri::ArcGISRuntime::SceneQuickView::mouseClicked, this,
+      [sceneView, this](QMouseEvent& mouseEvent)
+      {
+        setPointToConvert(sceneView->screenToBaseSurface(mouseEvent.x(), mouseEvent.y()));
+      });
+    }
+    else if (dynamic_cast<Esri::ArcGISRuntime::MapQuickView*>(geoView))
+    {
+      auto mapView = static_cast<Esri::ArcGISRuntime::MapQuickView*>(geoView);
+
+      setSpatialReference(mapView->spatialReference());
+      connect(mapView, &Esri::ArcGISRuntime::MapQuickView::spatialReferenceChanged, this,
+      [mapView, this]()
+      {
+        setSpatialReference(mapView->spatialReference());
+      });
+
+      connect(mapView, &Esri::ArcGISRuntime::MapQuickView::mouseClicked, this,
+      [mapView, this](QMouseEvent& mouseEvent)
+      {
+        setPointToConvert(mapView->screenToLocation(mouseEvent.x(), mouseEvent.y()));
+      });
+    }
```

Comment:
`spatialReferenceChanged` and `mouseClicked` signals should be moved to the new `ToolResourceProvider`